doc/texture_modifiers.adoc

```diff
@@ -0,0 +1,247 @@
+
+WARNING: Setting `w` or `h` to 0 will trigger a floating point exception, crashing the client.
+
+===== `<x>^[multiply:<color>`
+
+`color` is a `ColorString`. Each pixel of `x` will have it's RGB channels multiplied with those of `color`; the alpha value of `color` is ignored.
+// TODO reference
+
+===== `<x>^[colorize:<color>[:<ratio>]`
+
+Interpolates between `color`, a `ColorString`, and the pixel colors of `x` as specified by the `ratio`:
+
+* Defaults to the alpha of `color` if omitted;
+* If it is an integer from 0 (only pixel color) to 255 (only `color`), it is directly used as interpolation ratio: the resulting color of a pixel is `ratio` times `color` plus `(255 - ratio)` times pixel color;
+* If it is the string `alpha`, the texture pixel's alpha value determines the `ratio` per pixel
+
+===== `<x>^[mask:<texture>`
+
+The dimensions of the resulting texture are determined by the base texture. Applies bitwise and to all RGBA values of `texture` and `x`. If a pixel of `x` is out of bounds on `texture`, the value of `x` is preserved. Is associative and commutative if all used textures have the same dimensions.
+
+===== `<x>^[lowpart:<percent>:<texture>`
+
+Overlays the lower `percent` part of `texture` on the base texture `x`.
+
+TIP: Use `blank.png` as base texture `x` if you do not want a background
+
+==== Base Texture Generators
+
+These modifiers do not accept a base texture as they generate one from their arguments.
+
+===== `[png:<data>`
+
+Creates a texture from an embedded base64-encoded PNG image. The base64 string `data` can be produced by combining `minetest.encode_base64` and `minetest.encode_png`.
+
+WARNING: Do not use this for large textures. If used as an object texture, this will get sent arbitrarily often, putting a strain on the network.
+
+TIP: Consider using other texture modifiers cleverly or using dynamic media instead.
+
+===== `[combine:<w>x<h>:<textures>`
+
+A **combining** texture modifier accepting other texture modifiers as arguments.
+
+* `w`: Width of the resulting texture
+* `h`: Height of the resulting texture
+* `textures`: Colon (`:`)-separated list of locations and textures to blit in the form `<x>,<y>=<texture>`. Textures must be escaped if they contain texture modifiers. Nesting combine is possible through escaping. The list of textures to blit can be empty.
+
+Result: A texture of dimensions `w` times `h` on which all `textures` have been blit at the specified locations. The background is black and transparent.
+
+===== `[inventorycube{<top>{<left>{<right>`
+
+Renders a cube with the three given textures using simple software rendering. The resulting image will be 9 times the nearest power of 2 that is large enough to represent the contain the dimensions of the largest image, clamped to a range of at least 4 and at most 64.
```

Comment:
> the nearest power of 2 that is large enough to represent the contain the dimensions of the largest image

… what? I am confused by the sentence structure.

Reply:
Should be "~~the represent~~ to contain"